refactor(linter_codegen): simplify matching "or" patterns

overlookmotel opened the pull request:
Follow-on after #13138, as per [this comment](https://github.com/oxc-project/oxc/pull/13138#discussion_r2328545173).

Pure refactor. Break out of loop early on first failure, as following iterations have no effect.

camc314 commented:
### Merge activity

* **Sep 8, 3:19 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(linter_codegen): simplify matching "or" patterns (#13576)

Follow-on after #13138, as per [this comment](https://github.com/oxc-project/oxc/pull/13138#discussion_r2328545173).

Pure refactor. Break out of loop early on first failure, as following iterations have no effect.
